pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -89,6 +89,88 @@ func getApplicablePodNameForAppFramework(cr splcommon.MetaObject, ordinalIdx int
 	return fmt.Sprintf("splunk-%s-%s-%d", cr.GetName(), podType, ordinalIdx)
 }
 
+// runCustomCommandOnSplunkPods creates the required directory for the pod/s
+func runCustomCommandOnSplunkPods(ctx context.Context, cr splcommon.MetaObject, replicas int32, command string, podExecClient splutil.PodExecClientImpl) error {
+	var err error
+	var stdOut string
+
+	streamOptions := splutil.NewStreamOptionsObject(command)
+	// Run the command on each replica pod
+	for replicaIndex := 0; replicaIndex < int(replicas); replicaIndex++ {
+		// get the target pod name
+		podName := getApplicablePodNameForAppFramework(cr, replicaIndex)
+		podExecClient.SetTargetPodName(ctx, podName)
+
+		// CSPL-1639: reset the Stdin so that reader pipe can read from the correct offset of the string reader.
+		// This is particularly needed in the cases where we are trying to run the same command across multiple pods
+		// and we need to clear the reader pipe so that we can read the read buffer from the correct offset again.
+		splutil.ResetStringReader(streamOptions, command)
+
+		// Throw an error if we are not able to run the command
+		stdOut, _, err = podExecClient.RunPodExecCommand(ctx, streamOptions, []string{"/bin/sh"})
+		if err != nil {
+			err = fmt.Errorf("unable to run command %s. stdout: %s, err: %s", command, stdOut, err)
+			break
+		}
+	}
+	return err
+}
+
+// addTelApp adds a telemetry app
+func addTelApp(ctx context.Context, client splcommon.ControllerClient, replicas int32, cr splcommon.MetaObject) error {
+	var err error
+
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("addTelApp").WithValues(
+		"name", cr.GetObjectMeta().GetName(),
+		"namespace", cr.GetObjectMeta().GetNamespace())
+
+	// Create pod exec client
+	podExecClient := splutil.GetPodExecClient(client, cr, "")
+
+	// Handle non SHC scenarios(Standalone, CM, LM)
+	if cr.GetObjectKind().GroupVersionKind().Kind != "SearchHeadCluster" {
```

Comment:
1. Any reason, we can't use the bundle push mechanism for the IDXC also, that way, we don't need to call the reload end point for each indexer instance.
2. Don't we need the app on the deployer?
3. Also, considering a future possibility of having a separate CRD for the deployer, i think we can just check for the SHC explicitly.

Reply:
Hi @sgontla , as discussed the app will exist only on primary instances i.e CM and not the indexers. The app is on the deployer as well as the SHC members currently. When we have a separate CRD for deployer we can check with kind as "Deployer"